zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/adapters/ClientAdapter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.adapters;
+
+import io.camunda.exporter.ExporterResourceProvider;
+import io.camunda.exporter.config.ExporterConfiguration;
+import io.camunda.exporter.schema.SchemaManager;
+import io.camunda.exporter.schema.SearchEngineClient;
+import io.camunda.exporter.store.BatchRequest;
+import io.camunda.search.connect.configuration.ConnectConfiguration;
+import java.io.IOException;
+
+public interface ClientAdapter {
```

Comment:
To be honest I'm a massive fan of a static factory e.g. `ClientAdapter.of(config)` (or any appropriate name), this lets us hide details about whether it is an opensearch/elasticsearch in the config we can place the `createClientAdapter` logic in this `of` function